.github/workflows/test-integration-rosa.yaml

```diff
@@ -0,0 +1,352 @@
+          aws configure set aws_access_key_id ${{ secrets.DISTRO_CI_AWS_ACCESS_KEY }} --profile=${{ secrets.DISTRO_CI_AWS_PROFILE }}
+     
+      - name: Create ROSA cluster and login
+        uses: camunda/camunda-tf-rosa/.github/actions/rosa-create-cluster@main
+        timeout-minutes: 125
+        env:
+          AWS_PROFILE: ${{ secrets.DISTRO_CI_AWS_PROFILE }}
+        with:
+          rh-token: ${{ secrets.DISTRO_CI_REDHAT_CONSOLE_TOKEN }}
+          admin-username: ${{ secrets.DISTRO_CI_OPENSHIFT_CLUSTER_USERNAME }}
+          admin-password: ${{ secrets.DISTRO_CI_OPENSHIFT_CLUSTER_PASSWORD }}
+          s3-backend-bucket: "${{ env.TF_S3_BUCKET }}-${{ env.TEST_AWS_REGION }}"
+
+          cluster-name: ${{ fromJson(needs.read_cluster_vars.outputs.result).cluster_name[matrix.distro.name] }}
+          aws-region: ${{ env.TEST_AWS_REGION }}
+          replicas: 4
+
+          openshift-version: "${{ matrix.distro.version }}"
+
+      - name: Configure on-demand cluster ROSA
+        timeout-minutes: 10
+        # we need to retry as CRDs can take some time to be installed
+        uses: nick-fields/retry@7152eba30c6575329ac0576536151aca5a72780e # v3
+        with:
+          timeout_minutes: 10
+          max_attempts: 40
+          shell: bash
+          retry_wait_seconds: 15
+          command: |
+              cd ./.github/workflows/fixtures/clusters/rosa-hcp-on-demand/
+          
+              : # populate with secrets then kustomize and apply!
+              export EXTERNAL_DNS_GCP_SERVICE_ACCOUNT=${{ secrets.DISTRO_CI_ON_DEMAND_EXTERNAL_DNS_GCP_SERVICE_ACCOUNT }}
+              export CERT_MANAGER_GCP_SERVICE_ACCOUNT=${{ secrets.DISTRO_CI_ON_DEMAND_CERT_MANAGER_GCP_SERVICE_ACCOUNT }}
+              envsubst < cert-manager/secret.yaml.tpl > cert-manager/secret.yaml
+              envsubst < external-dns/secret.yaml.tpl > external-dns/secret.yaml
+              kustomize build --enable-helm ./ | kubectl apply -f - 
+                  
+              echo "Waiting for global state to converge"
+              sleep 30
+                  
+              while kubectl get pods --all-namespaces | grep -E -q -v '(Running|Completed|STATUS)'; do
+                  echo "Waiting for all pods to be Running or Completed"
+                  sleep 5
+              done
+          
+              kubectl get all --all-namespaces
+      
+      - name: Export kubeconfig and encrypt it # this is required to pass matrix outputs securely using artifacts
+        id: export_kube_config
+        run: |
+          echo "$(kubectl config view --raw)" > kubeconfig.yaml 2>/dev/null
+          openssl enc -aes-256-cbc -salt -in kubeconfig.yaml -out encrypted_kubeconfig.enc -pass pass:"${GITHUB_TOKEN}" -pbkdf2
+          encrypted_kubeconfig_base64=$(base64 -w 0 encrypted_kubeconfig.enc)
+          echo "kubeconfig_raw=${encrypted_kubeconfig_base64}" >> "$GITHUB_OUTPUT"
+
```

Comment:
interesting choice of implementation.
Just curious how difficult it is to create kubeconfigs on demand via the rosa client?